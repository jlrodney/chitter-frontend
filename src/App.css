.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(15px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
.Peeps-container {
  text-align: center;
  object-fit: cover;
  display: inline-grid;
  border-style: double;
  border-width: 15px;
  box-shadow: 5px 5px 5px black;
  transition: filter 0.3s;
  width: 600px;
  overflow: hidden;
}

/* .form {
  background-color: #3CBC8D;
  color: white;
} */
input[type=text], input[type=password] {
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    display: inline-block;
    border: 1px solid #ccc;
    box-sizing: border-box;
}
button {
    background-color: #4CAF50;
    color: white;
    padding: 14px 20px;
    margin: 8px 0;
    border: none;
    cursor: pointer;
    width: 100%;
    font-family: Arial, Helvetica, sans-serif;
}

button:hover {
    opacity: 0.8;
}


* {
  box-sizing: border-box
}

html {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: 'Dosis', sans-serif;
  line-height: 1.6;
  color: #666;
  background: #F6F6F6;
}

#root {
  max-width: 1200px;
  margin: 0 auto;
}


img {
  display: block;
  margin: 1rem auto;
  max-width: 100%;
}

p {
  padding: 0 2.5rem 2.5rem;
  margin: 0;
}

.container {
  display: flex;
  flex-wrap: wrap;
}

.card {
  margin: 1rem;
  background: white;
  box-shadow: 2px 4px 25px rgba(0, 0, 0, .1);
  border-radius: 12px;
  overflow: hidden;
  transition: all .2s linear;
}

.card:hover {
  box-shadow: 2px 8px 45px rgba(0, 0, 0, .15);
  transform: translate3D(0, -2px, 0);
}

@media screen and (min-width: 200px) {
  .card {
    flex: 1 1 calc(50% - 2rem);
  }
}

@media screen and (min-width: 200px) {
  .card {
    flex: 1 1 calc(33% - 2rem);
  }
}
