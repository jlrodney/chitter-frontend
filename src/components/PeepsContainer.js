import React, { Component } from 'react';
// import axios from 'axios';
import Peep from './Peep';
import NewPeepForm from './NewPeepForm';
import NewUserForm from './NewUserForm'
import SignInForm from './SignInForm';


const GET_PEEPS_URL = "https://chitter-backend-api.herokuapp.com/peeps"
const CREATE_USER_URL = 'https://chitter-backend-api.herokuapp.com/users';
const CREATE_PEEPS_URL = 'https://chitter-backend-api.herokuapp.com/peeps';
const CREATE_SESSION_URL = 'https://chitter-backend-api.herokuapp.com/sessions';
const DELETE_PEEP_URL = 'https://chitter-backend-api.herokuapp.com/peeps/'
let sessionKey = ''
let userID = '';

class PeepsContainer extends Component {

  constructor(props){
    super(props)
    this.state = {
      peeps: [],
    }
    this.addNewPeep = this.addNewPeep.bind(this)
    this.removePeep = this.onRemovePeep.bind(this)
    this.addNewUser = this.addNewUser.bind(this)
    this.userSignIn = this.userSignIn.bind(this)
  }

  componentDidMount() {
    this.getPeeps();
  }

  async getPeeps() {
    fetch(GET_PEEPS_URL)
    .then((response) => response.json())
    .then((data) => {
      // console.log(data)
      this.setState({
        peeps: data
      })
      console.log(this.state.peeps)
    })
    .catch(error => console.log(error))
  }

  async addNewPeep(body) {
    let data = { peep: { user_id: userID, body: body } };
    let response = await fetch(CREATE_PEEPS_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json; charset=utf-8',
        Authorization: `Token token=${sessionKey}`
      },
      body: JSON.stringify(data)
    });
    console.log(JSON.stringify(response))
    this.getPeeps();
  }

  async onRemovePeep(id) {
    let response = await fetch(DELETE_PEEP_URL + id, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json; charset=utf-8',
        Authorization: `Token token=${sessionKey}`
      }
    });
    console.log(JSON.stringify(response))
    this.getPeeps();
  }

  async addNewUser(handle, password) {
      let data = { user: { handle: handle, password: password } };
      let response = await fetch(CREATE_USER_URL, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json; charset=utf-8'
        },
        body: JSON.stringify(data)
      });
      return await response.json();
    }

  async userSignIn(handle, password) {
      let data = {"session": {"handle":handle, "password":password}};
      let response = await fetch(CREATE_SESSION_URL, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json; charset=utf-8'
        },
        body: JSON.stringify(data)
      });
      let sessionData = await response.json();
      console.log(JSON.stringify(response))
          sessionKey = sessionData.session_key;
          userID = sessionData.user_id;
          console.log(userID)
          console.log(sessionKey)
          console.log(sessionData)
          return sessionData;
    }

render() {
  return (
    <div className="forms">
          <NewUserForm onNewUser={this.addNewUser}/>
          <SignInForm userSignIn={this.userSignIn}/>
          <NewPeepForm onNewPeep={this.addNewPeep}/>

        <div className="Peeps-container">
        {this.state.peeps.map( peep => {
          return (<Peep
              peep={peep}
              key={peep.id}
              onRemovePeep={this.removePeep}
            />)
            }
          )}
        </div>
      </div>
    )
  }
}

export default PeepsContainer;
