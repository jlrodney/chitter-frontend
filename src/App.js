
import React, { Component } from 'react';
// eslint-disable-next-line
import logo from './logo.svg';
import './App.css';
import PeepsContainer from './components/PeepsContainer';


class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Chitter Challenge!</h1>
        </header>
        <PeepsContainer />
      </div>
    );
  }
}

export default App;
