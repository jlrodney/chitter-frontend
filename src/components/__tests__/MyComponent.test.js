
import React from 'react';
import { shallow, mount } from 'enzyme';
import  App from '../../App.js';
import PeepsContainer from '../PeepsContainer.js'

describe('should render title', () => {

  it('should render correctly with no props', () => {
    const component = shallow(<App/>);
    expect(component).toMatchSnapshot();
  });


  it('should have title', () => {
    const component = shallow(<App/>);
    var node = component.find('h1')
    expect(node.hasClass('App-title')).toBeTruthy();
    expect(node.html()).toEqual('<h1 class="App-title">Chitter Challenge!</h1>')
  });
})

describe('should render posts', () => {
  it('creates todo', () => {
  const wrapper = shallow(<PeepsContainer />);
  wrapper.find('input').simulate('change', {
    target: { value: 'hello' }
  })
  wrapper.find('button').simulate('click');

  });

  it('can add peep', () => {
    const message = 'hmmm'
    const component = mount(<App/>);
    var node = component.find('.body').setValue(message)
    component.find('.submess').onClick()
    const actual = component.find('.Peeps-container').getText()
    expect(actual).to.equal(message)
  })
})
