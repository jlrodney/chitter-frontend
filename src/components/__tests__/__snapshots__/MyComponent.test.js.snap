// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render title should render correctly with no props 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Chitter Challenge!
    </h1>
  </header>
  <PeepsContainer />
</div>
`;
